torch/onnx/_internal/exporter.py

```diff
@@ -350,16 +351,31 @@ class ResolvedExportOptions(ExportOptions):
 
     @_beartype.beartype
     def __init__(
-        self, options: Optional[Union[ExportOptions, "ResolvedExportOptions"]]
+        self,
+        options: Optional[Union[ExportOptions, "ResolvedExportOptions"]] = None,
+        model: Optional[Union[torch.nn.Module, Callable, torch_export.ExportedProgram]] = None,  # type: ignore[name-defined]
     ):
-        if options is None:
-            options = ExportOptions()
+        from torch.onnx._internal.fx import (  # TODO: Prevent circular dep
+            diagnostics,
+            dynamo_graph_extractor,
+            torch_export_graph_extractor,
+        )
+
+        assert options is not None, "ExportOptions must be specified."
```

Comment:
why this change?

Reply:
We were initializing `ExportOptions` in different points of the code. I found out this when writing a unit test that would detect an invalid `fx_tracer` and `model` combination. With this change, `def dynamo_export` is responsible to initialize `options` in a single spot

Reply:
The original code looks clean too as all the None check are postponed to this point inside `ResolvedExportOptions`.

I don't have a strong opinion. If you are continuing with this change, please update argument `options: Optional[Union[ExportOptions, "ResolvedExportOptions"]] = None,` to `options: Union[ExportOptions, "ResolvedExportOptions"]`. Doesn't make sense to allow None that will immediately trigger assertion error.